### PR TITLE
Add flake.nix

### LINKED ISSUE #61: Provide Nix packages
I have no experience with Nix yet, so I'd appreciate help with this.

### PR DESCRIPTION
**Pull request by pingiun**
Fixes #61 

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

**Comment by pingiun** [comment]
paging @colemickens and @arxanas for review

File: flake.nix
Comment (arxanas) [inline flake.nix:2]: This project is called `jujutsu`, please update across this file. (`jujitsu` is also a common spelling, but not the right one here.)

File: flake.lock
Comment (arxanas) [inline flake.lock:1]: Is this a strictly-generated file? If so, it might be good to add a `.gitattributes` file and mark it as such: https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github

File: .gitignore
Comment (arxanas) [inline .gitignore:2]: Can you add a comment explaining that these entries would be generated by Nix?

File: flake.nix
Comment (arxanas) [inline flake.nix:103]: Should this run `cargo test`?

File: flake.nix
Comment (arxanas) [inline flake.nix:88]: Is this to support a version of Rust which isn't available in Nix? @martinvonz might be open to lowering the minimum-supported Rust version instead.

**Review by arxanas — COMMENTED** [review]
Can you add a CI job to verify the Nix build? Example here: https://github.com/arxanas/git-branchless/blob/master/.github/workflows/nix-linux.yml. (`git-branchless` would break the Nix build all the time if not for that.)

File: flake.nix
Comment (pingiun) [reply flake.nix:103]: This is the default for a derivation generated by rustPlatform.buildRustPackage. I think that works via hooks

File: flake.nix
Comment (pingiun) [reply flake.nix:88]: I think nightly rust is required for this project, which isn't packaged in nixpkgs. I use the mozilla nixpkgs overlay. This requires a sha256 hash or rustup manifest file to install a rust channel hermetically. I will add the ./toolchain-manifest.toml to the gitattributes too

Comment (pingiun) [reply flake.nix:88]: I will also explain with some comments how to use the script

Comment (martinvonz) [reply flake.nix:103]: I'm fine with making the project compatible with a stable Rust version if that helps. I won't have access to a computer until a week from now, so I can't make the changes myself, but I can approve PRs if anyone else wants to do it.

**Comment by martinvonz** [comment]
> Fixes #61

Thanks!

> - [ ] Appropriate changes to README are included in PR

 I guess this is about adding a note to the installation instructions? Makes sense.

Anything else I should wait for before merging the PR? Were @arxanas and @colemickens happy with the review? (Sorry if that's indicated somewhere. I'm new to collaborating on GitHub, and especially the mobile UI.)

**Review by arxanas — APPROVED** [review]
LGTM assuming CI passes.

**Comment by colemickens** [comment]
flake.nix looks good, I think my only feedback would be to try to patch the version so that this is more accurate:

```
❯ nix run github:martinvonz/jj/9fb580b6eeba2bbd79658f9777000cabb2ebdaed -- --version
jj 0.2.0 
```

Ideally that might look like:
```
❯ nix run github:martinvonz/jj/9fb580b6eeba2bbd79658f9777000cabb2ebdaed -- --version
jj unstable-9fb580b6
```

or something like that. Not a blocker though! Love to see it, Nix ftw!

**Pull request merged**
